src/ProjectMWrapper.cpp

```diff
@@ -59,6 +63,48 @@ void ProjectMWrapper::initialize(Poco::Util::Application& app)
 
         _projectM = projectm_create_settings(&settings, PROJECTM_FLAG_NONE);
 
+        // remove any presets that don't include filter
+        if (presetFilter != "") {
+            SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Preset include filter '%s'\n", presetFilter.c_str());
+
+            unsigned int size = projectm_get_playlist_size(_projectM);
+            for (int i = size; i > 0; i--) {
+                std::string presetName(projectm_get_preset_name(_projectM, i-1));
+                unsigned int presetIndex = projectm_get_preset_index(_projectM, presetName.c_str());
+                std::string presetFileName(projectm_get_preset_filename(_projectM, presetIndex));
+
+                if (presetFileName.find(presetFilter) == std::string::npos) {
+                    projectm_remove_preset(_projectM, presetIndex);
+                } else {
+                    SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Preset include '%s'\n", presetFileName.c_str());
+                }
+            }
+        }
+
+        // remove any presets in exclude.txt
+        Poco::Path excludeFile = Poco::Path(presetExcludeFile);
+        if (Poco::File(excludeFile).exists()) {
+            SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Preset exclude config '%s'\n", excludeFile.toString().c_str());
+
+            // TODO: use Poco to read file?
```

Comment:
is it important to use Poco for cross platform portability? I tried for a while but wasn't able to figure out how to read/write files. I couldn't find many examples of using this framework.

Reply:
Poco::File isn't for file I/O, but to query file information like size, existence and other things. These are quite platform-specific, so Poco is a great tool for this. We could use `std::filesystem`, but this is only supported since C++17, previous C++ versions would require Boost (as used in libprojectM's new playlist library). Since Poco is already used extensively, this additional dependency is not really required as Poco::File implements the requires features.